src/zope/app/apidoc/typemodule/type.py

```diff
@@ -69,10 +69,11 @@ def items(self):
         """See zope.container.interfaces.IReadContainer"""
         results = [(name, LocationProxy(iface, self, name))
                    for name, iface in getUtilitiesFor(self.interface)]
-        results.sort(lambda x, y: cmp(x[1].getName(), y[1].getName()))
+        results.sort(key=lambda x: x[1].__name__)
```

Comment:
Ah, nope: the same test will fail on master with `AttributeError: Foo instance has no attribute 'getName'` if I change the `cmp` function to a `key=` function. And that's because a `cmp` function *isn't called at all* for a list with only one element, whereas the `key` function *is called*:

```pycon
>>> [1].sort(lambda x, y: x.NOATTR)
None
>>> [1].sort(key=lambda x: x.NOATTR)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 1, in <lambda>
AttributeError: 'int' object has no attribute 'NOATTR'
```